dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java

```diff
@@ -738,7 +738,9 @@ private void checkCompatibleTypeAndSubclass(Workflow existing, Workflowish updat
                     String.format("You can't add a %s version to a %s %s, the descriptor language of all versions must be the same.", update.getSubclass(), existing.getDescriptorType(), existingTerm));
             }
         } else if (existingType == EntryType.SERVICE) {
-            if (existing.getDescriptorTypeSubclass() != toDescriptorTypeSubclass(update.getSubclass().toString())) {
+            final DescriptorLanguageSubclass serviceSubClass = toDescriptorTypeSubclass(update.getSubclass().toString());
+            // Services created prior to 1.14 have the descriptorTypeSubclass value of "n/a". #5636
+            if (existing.getDescriptorTypeSubclass() != DescriptorLanguageSubclass.NOT_APPLICABLE && existing.getDescriptorTypeSubclass() != serviceSubClass) {
```

Comment:
Other options:

1. Update the existing services' `descriptorTypeSubclass` values away from `n/a`. It's hard to tell from pure SQL though -- you have to read the .dockstore.yml to get the subclass. We could add an endpoint.
2. Don't do this at all, and force existing services that are getting updated to set the subclass set to `n/a`.

There aren't that many services; only members of the Dockstore team have created them AFAIK, maybe we're putting too much work into worrying about existing services.

The other idea is when we detect it's `n/a`, we update it to the subclass of the new version. But `n/a` can be a [valid value](https://github.com/dockstore/dockstore/blob/8ba6457b53d3343e71913167c5bb819bc393b259/dockstore-common/src/main/java/io/dockstore/common/DescriptorLanguageSubclass.java#L45), so...